src/main/java/io/nats/client/support/NatsRequestCompletableFuture.java

```diff
@@ -6,11 +6,14 @@
 import java.time.Duration;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
 
 /**
  * This is an internal class and is only public for access.
  */
 public class NatsRequestCompletableFuture extends CompletableFuture<Message> {
+    public static final boolean USE_TIMEOUT_EXCEPTION = Boolean.getBoolean(System.getProperty("nats.use-timeout-exception")); // TODO: Remove. Use TimeoutException instead of CancellationException.
```

Comment:
Can we add `nats.use-timeout-exception` in the Option class
```
public static final String PROP_USE_TIMEOUT_EXCEPTION = PFX + "use.timeout.exception";
```
This will also mean that you will have to modify the `NatsRequestCompletableFuture` constructor to receive the setting from the connection and then change `NatsConnection` line 1170

Reply:
done. however, it would be nice if you labeled these changes as temporary to support legacy. in the next version, it's preferable to use TimeoutException without additional questions, IMO.

Reply:
I'm okay with adding a comment to make this the default in future versions.